udata/tests/api/test_dataservices_api.py

```diff
@@ -151,6 +157,52 @@ def test_dataservice_api_create(self):
         self.assertEqual(dataservice.title, "Undeleted title")
         self.assertIsNone(dataservice.deleted_at)
 
+    def test_dataservice_api_list_owned(self) -> None:
+        """Should filters out private dataservices if not owner"""
+        owner = UserFactory()
+        org = OrganizationFactory()
+        public_dataservice = DataserviceFactory()
+        private_user_dataservice = DataserviceFactory(private=True, owner=owner)
+        private_org_dataservice = DataserviceFactory(private=True, organization=org)
+
+        # Only public dataservices for non-authenticated user.
+        response = self.get(url_for("api.dataservices"))
+        assert200(response)
+        assert len(response.json["data"]) == 1
+        assert dataservice_in_response(response, public_dataservice)
+
+        # Only public dataservices for a non-owner authenticated user.
+        self.login(UserFactory())
+        response = self.get(url_for("api.dataservices"))
+        assert200(response)
+        assert len(response.json["data"]) == 1
+        assert dataservice_in_response(response, public_dataservice)
+
+        # Authenticated user is the owner
+        self.login(owner)
+
+        # Public and 1 private dataservice for the owner
+        response = self.get(url_for("api.dataservices"))
+        assert200(response)
+        assert len(response.json["data"]) == 2  # Return everything
+        assert dataservice_in_response(response, public_dataservice)
+        assert dataservice_in_response(response, private_user_dataservice)
+
+        # Authenticated user is now also member of the organization
+        member = Member(user=owner, role="editor")
+        org.members = [member]
+        org.save()
+        del owner.organizations  # clear user.organizations cached property
```

Comment:
Wow didn't know we had to do that 🤯

Reply:
Well, I had to debug to understand what was happening :scream: